tf2onnx/onnx_opset/tensor.py

```diff
@@ -1097,36 +1098,76 @@ def version_1(cls, ctx, node, **kwargs):
 
-        if len(input_shape) == 3:
-            # add a squeeze op to convert output into 3d
-            kwargs = {**inputs_map}
-            ctx.remove_node(node.name)
-            slice1 = GraphBuilder(ctx).make_slice(kwargs)
-            ctx.make_node("Squeeze", [slice1], {"axes": [3]},
-                          outputs=node.output, name=node.name, dtypes=dtypes, shapes=shapes)
+        if ctx.get_node_by_output(node.input[2]).is_const():
+            crops = node.inputs[2].get_tensor_value()
+            # implement crop logic, the data format is NHWC
+            slice_axis = [1, 2]
+            top, bottom = crops[0]
+            left, right = crops[1]
+            starts = [top, left]
+            ends = []
+            for end in [bottom, right]:
+                if end != 0:
+                    ends.append(-end)
+                else:
+                    ends.append(np.iinfo(np.int32).max)
+
+            attr = {"axes": slice_axis, "ends": ends, "starts": starts}
+            inputs_map = {"data": trans2.output[0], **attr}
+
+            if len(input_shape) == 3:
+                # add a squeeze op to convert output into 3d
+                kwargs = {**inputs_map}
+                ctx.remove_node(node.name)
+                slice1 = GraphBuilder(ctx).make_slice(kwargs)
+                ctx.make_node("Squeeze", [slice1], {"axes": [3]},
+                              outputs=node.output, name=node.name, dtypes=dtypes, shapes=shapes)
+            else:
+                kwargs = {**inputs_map, "outputs": node.output}
+                ctx.remove_node(node.name)
+                GraphBuilder(ctx).make_slice(kwargs, name=node.name, dtypes=dtypes, shapes=shapes)
         else:
+            crops_type = ctx.get_dtype(node.input[2])
+            crops_np_type = utils.map_onnx_to_numpy_type(crops_type)
+            # Make the node crops being transposed and respectively obtained its two rows by split op
+            crops_trans = ctx.make_node("Transpose", node.inputs[2].output, {"perm": [1, 0]})
+            trans_split = ctx.make_node("Split", [crops_trans.output[0]], output_count=2, attr={"axis": 0})
+            starts = ctx.make_node("Squeeze", [trans_split.output[0]], attr={"axes": [0]})
+            # Create the original node for ends
+            ends_ori = ctx.make_node("Squeeze", [trans_split.output[1]], attr={"axes": [0]})
+            # create zero const node
+            zero_value = np.array([0]).astype(crops_np_type)
+            zero_const = ctx.make_const(utils.make_name("Const"), zero_value)
+            # Obtain a node by comparing the ends_ori with zero
+            ends_ori_equal_zero = ctx.make_node("Equal", [ends_ori.output[0], zero_const.output[0]])
+            ends_ori_equal_zero = ctx.make_node("Cast", [ends_ori_equal_zero.output[0]], attr={"to": crops_type})
+            # Create int max const node
+            int_max_value = np.array([utils.get_max_value(crops_np_type)]).astype(crops_np_type)
```

Comment:
do we need astype(crops_np_type) here? I think utils.get_max_value(crops_np_type) will return the correct dtype

Reply:
Thanks. I thinks it is needed. Without astype(crops_np_type), there will be an error about unmatched type(int32 and int64).